pkg/intermediate/aggregate.go

```diff
@@ -148,37 +169,97 @@ func (a *AggregationProcess) ForAllRecordsDo(callback FlowKeyRecordMapCallBack)
 	return nil
 }
 
-// GetLastUpdatedTimeOfFlow provides the last updated time in the format of IPFIX
-// field "flowEndSeconds".
-func (a *AggregationProcess) GetLastUpdatedTimeOfFlow(flowKey FlowKey) (uint32, error) {
+func (a *AggregationProcess) deleteFlowKeyFromMap(flowKey util.FlowKey) error {
 	a.mutex.Lock()
 	defer a.mutex.Unlock()
-	record, exists := a.flowKeyRecordMap[flowKey]
+	return a.deleteFlowKeyFromMapWithoutLock(flowKey)
+}
+
+func (a *AggregationProcess) deleteFlowKeyFromMapWithoutLock(flowKey util.FlowKey) error {
+	_, exists := a.flowKeyRecordMap[flowKey]
 	if !exists {
-		return 0, fmt.Errorf("flow key is not present in the map")
+		return fmt.Errorf("flow key is not present in the map")
 	}
-	flowEndField, exists := record.Record.GetInfoElementWithValue("flowEndSeconds")
-	if exists {
-		return flowEndField.Value.(uint32), nil
-	} else {
-		return 0, fmt.Errorf("flowEndSeconds field is not present in the record")
+	delete(a.flowKeyRecordMap, flowKey)
+	return nil
+}
+
+// GetExpiryFromExpirePriorityQueue returns the earliest timestamp (active expiry
+// or inactive expiry) from expire priority queue.
+func (a *AggregationProcess) GetExpiryFromExpirePriorityQueue() time.Duration {
+	a.mutex.Lock()
+	defer a.mutex.Unlock()
+
+	if a.expirePriorityQueue.Len() > 0 {
+		topItem := a.expirePriorityQueue.Peek()
+		if topItem.ActiveExpireTime.Before(topItem.InactiveExpireTime) {
+			return topItem.ActiveExpireTime.Sub(time.Now())
+		} else {
+			return topItem.InactiveExpireTime.Sub(time.Now())
+		}
+	}
+	if a.activeExpiryTimeout < a.inactiveExpiryTimeout {
+		return a.activeExpiryTimeout
 	}
+	return a.inactiveExpiryTimeout
```

Comment:
Is the definition of activeExpiryTimeout and inactiveExpiryTimeout described somewhere or in a doc?
I don't quiet understand their meanings from line 62 and 65.

In this GetExpiryFromExpirePriorityQueue function, it looks like the caller will not know if the return value is from activeExpiryTimeout or inactiveExpiryTimeout. 
Does that mean active or inactive is not important?
So do we need to store them separately? 
For example, can we just store the smaller one during the initialization if it is the one being returned here?

Reply:
Please refer this: https://github.com/vmware-tanzu/antrea/pull/1949/files#diff-6cf2c6fa47e75165fd7d385fc7ff6cce0a2eda02cfb7cf04db849031ae126ba9R134-R147
I will also add more context to the existing text in field descriptions.

A flow record can be expired because of either active or inactive flow timeouts. They do not have a constant relationship between them. Depending on timestamps of flow records received, these active and inactive timeouts get set/reset, so tracking as part of the initialization will not work.